### PR TITLE
Unit tests for v1.0.0

### PR DESCRIPTION
**Pull request by devahmedshendy**

### CODE REVIEW

File: Sources/Contribute/URL.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  /// Creates a temporary file from the given content.
+  ///
+  /// - Parameter content: The content of the temporary file.
+  /// - Returns: The URL of the created temporary file.
+  /// - Throws: An error if the temporary file creation fails.
+  public static func temporaryFile(fromContent content: String) throws -> URL {
+    let temporaryFileURL = temporaryDirURL.appendingPathComponent(UUID().uuidString)
+    try content.write(to: temporaryFileURL, atomically: true, encoding: .utf8)
+    return temporaryFileURL
+  }
```
Comment (leogdion) [inline Sources/Contribute/URL.swift]: Why did you move this?

File: Tests/ContributeTests/testError.swift [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import Foundation
+
+let testError = NSError(domain: "TestErrorDomain", code: 0)
```
Comment (leogdion) [inline Tests/ContributeTests/testError.swift]: What is this for?

File: Tests/ContributeTests/Spies/NetworkManagerSpies.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  }
+}
+
+
+internal final class NetworkManagerFailedSpy: URLNetworkManager {
+  func download(
+    fromURL: URL,
+    completion: @escaping @Sendable (URL?, URLResponse?, Error?) -> Void
+  ) {
+    print("HERE")
```
Comment (leogdion) [inline Tests/ContributeTests/Spies/NetworkManagerSpies.swift]: remove this

File: Tests/ContributeTests/Spies/FrontMatterFormatterSpy.swift [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import Foundation
+import Contribute
+
+final class FrontMatterFormatterSpy: FrontMatterFormatter {
+  func format<MockFrontMatter>(_ frontMatter: MockFrontMatter) throws -> String {
+    return "String"
```
Comment (leogdion) [inline Tests/ContributeTests/Spies/FrontMatterFormatterSpy.swift]: I would have it take in a string so you know what to expect

File: Tests/ContributeTests/Spies/FrontMatterExporterSpies.swift [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import Foundation
+import Contribute
+
+internal final class FrontMatterExporterSuccessfulSpy: FrontMatterExporter {
```
Comment (leogdion) [inline Tests/ContributeTests/Spies/FrontMatterExporterSpies.swift]: I would create one Class and have take a Result property for which result you want

**Review by leogdion — COMMENTED** [review]
Great Work

File: Tests/ContributeTests/FileURLDownloaderTests.swift [hunk 6]
```diff
@@ -0,0 +1,237 @@
+// swiftlint:disable file_length
+// swiftlint:disable type_body_length
+// swiftlint:disable trailing_closure
+// swiftlint:disable function_body_length
```
Comment (devahmedshendy) [inline Tests/ContributeTests/FileURLDownloaderTests.swift]: @leogdion are we fine about this?

File: Tests/ContributeTests/YAMLTests.swift [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    )
+  }
+
+  internal func testValidTimezone() {
+    XCTAssertEqual(
+      YAML.dateFormatter.timeZone,
+      TimeZone.current
+    )
+  }
+}
```
Comment (devahmedshendy) [inline Tests/ContributeTests/YAMLTests.swift:18]: @leogdion are these all needed tests for `YAML.dateFormatter`?

File: Sources/Contribute/FileURLDownloader.swift [hunk 8]
```diff
@@ -56,7 +59,7 @@ public struct FileURLDownloader: URLDownloader {
     allowOverwrite: Bool,
     _ completion: @escaping (Error?) -> Void
   ) {
-    let task = session.downloadTask(with: fromURL) { destination, _, error in
+    _ = networkManager.download(fromURL: fromURL) { destination, _, error in
```
Comment (leogdion) [inline Sources/Contribute/FileURLDownloader.swift]: why the underscore? you shouldn't need that

File: Tests/ContributeTests/YAMLTests.swift (hunk 7 above)
Comment (leogdion) [reply Tests/ContributeTests/YAMLTests.swift:18]: Have you looked at the code coverage in Xcode or codecov?

File: Tests/ContributeTests/FileURLDownloaderTests.swift (hunk 6 above)
Comment (leogdion) [reply Tests/ContributeTests/FileURLDownloaderTests.swift]: No is it necessary?

File: Tests/ContributeTests/YAMLTests.swift (hunk 7 above)
Comment (devahmedshendy) [reply Tests/ContributeTests/YAMLTests.swift:18]: Oh I see, yeah it is shown as all done.

**Review by leogdion — APPROVED** [review]
